python/pyspark/util.py

```diff
@@ -183,6 +325,8 @@ def __del__(self):
 
 if __name__ == "__main__":
     import doctest
-    (failure_count, test_count) = doctest.testmod()
-    if failure_count:
-        sys.exit(-1)
+
+    if "pypy" not in platform.python_implementation().lower() and sys.version_info[:2] >= (3, 7):
```

Comment:
@viirya, let me just skip the tests here. Seems like it's tricky to conditionally skip the tests.